src/codec/framed_write.rs

```diff
@@ -193,12 +193,12 @@ where
                 match self.next {
                     Some(Next::Data(ref mut frame)) => {
                         tracing::trace!(queued_data_frame = true);
-                        let mut buf = (&mut self.buf).chain(frame.payload_mut());
-                        ready!(Pin::new(&mut self.inner).poll_write_buf(cx, &mut buf))?;
+                        let mut buf = (&mut self.buf).chain(frame.payload_mut()).to_bytes();
```

Comment:
After discussing with @seanmonstar I think in this case actually want to copy the frame payload onto the `buf` type.

Reply:
Probably via https://docs.rs/bytes/0.5.6/bytes/struct.BytesMut.html#method.extend_from_slice